src/api/apiService.js

```diff
@@ -1,15 +1,54 @@
-import { DEPARTMENT_BY_ID_URL, DEPARTMENTS_URL, LOGIN } from '../constants/url';
+import {
+  DEPARTMENT_BY_ID_URL,
+  DEPARTMENTS_URL,
+  LOGIN_URL,
+  LOGOUT_URL,
+  EMPLOYEES_URL,
+} from '../constants/url';
 import { api } from './interceptors';
 
 export default class apiService {
-  getDepartments() {
-    return api.get(DEPARTMENTS_URL);
+  getDepartments(limit = 10, page = 1) {
+    return api.get(`${DEPARTMENTS_URL}?limit=${limit}&page=${page}`);
   }
-  getDepartmentById(id) {
-    return api.get(`${DEPARTMENT_BY_ID_URL}${id}`);
+  getDepartment(id) {
+    return api.get(`${DEPARTMENTS_URL}/${id}`);
+  }
+  getSearchingDepartment(search) {
+    return api.get(`${DEPARTMENTS_URL}?q=${search}`);
+  }
+  getSortedDepartments(order) {
+    return api.get(`${DEPARTMENTS_URL}?sort=${order}`)
+  }
+  getEmployeesByDepartmentId(id) {
+    return api.get(`${DEPARTMENT_BY_ID_URL}${id}${EMPLOYEES_URL}`);
+  }
```

Comment:
apiService has all endpoints combined together. Much better to separate them by separated context related files